carbon-design-system/carbon-addons-iot-react
# feat(hierarchy-list): call onExpandedChange after rows are expanded

Closes #3191 

**Summary**

- I'm using an `onAfterExpand` callback to allow the HierarchyList to know when the items have all been expanded. I've added a new story to demonstrate an "Open all" behavior. I don't _love_ it, but I can't come up with a better solution. Additionally, when using the VirtualList, because of how the virtual list only renders what is currently visible, we can't (easily) know what the last element is, so I've left the previous code in place to fire the onExpandedChange event when the prop is updated when using VirtualList, because with the virtual list it happens almost instantly anyway.

**Change List (commits, features, bugs, etc)**

- pass onAfterExpand to list components
- pass isVirtualList to ListItem
- add new story to demonstrate an "Open all" behavior
- update tests to mock timer/requestAnimationFrame
- 

**Acceptance Test (how to verify the PR)**

- check the new [with a large number of expandable items story](https://deploy-preview-3281--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-large-number-of-expandable-items)
- click expand all button
- change to Actions tab
- click "Open all" button
- confirm the "onExpandedChange" function isn't call until the dom is updated

- repeat with `isVirtualList` knob

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
